metadataproxy/roles.py

```diff
@@ -157,117 +157,129 @@ def find_container(ip):
     return None
 
 
-def check_role_name_from_ip(ip, requested_role):
-    role_name = get_role_name_from_ip(ip)
-    if role_name == requested_role:
-        log.debug('Detected Role: {0}, Requested Role: {1}'.format(
-            role_name, requested_role
-        ))
-        return True
-    return False
-
-
 @log_exec_time
-def get_role_name_from_ip(ip, stripped=True):
+def get_role_params_from_ip(ip, requested_role=None):
+    params = dict(name=None, account_id=None, external_id=None, session_name=None)
+    role_name = None
     if app.config['ROLE_MAPPING_FILE']:
-        return ROLE_MAPPINGS.get(ip, app.config['DEFAULT_ROLE'])
-    container = find_container(ip)
-    if container:
-        env = container['Config']['Env']
-        for e in env:
-            key, val = e.split('=', 1)
-            if key == 'IAM_ROLE':
-                if val.startswith('arn:aws'):
-                    m = RE_IAM_ARN.match(val)
-                    val = '{0}@{1}'.format(m.group(2), m.group(1))
-                if stripped:
-                    return val.split('@')[0]
-                else:
-                    return val
-        msg = "Couldn't find IAM_ROLE variable. Returning DEFAULT_ROLE: {0}"
-        log.debug(msg.format(app.config['DEFAULT_ROLE']))
-        if stripped:
-            return app.config['DEFAULT_ROLE'].split('@')[0]
+        role = ROLE_MAPPINGS.get(ip, app.config['DEFAULT_ROLE'])
+        if isinstance(role, dict):
+            params.update(role)
         else:
-            return app.config['DEFAULT_ROLE']
+            role_name = role
     else:
-        return None
+        container = find_container(ip)
+        if container:
+            env = container['Config']['Env'] or []
+            # Look up IAM_ROLE and IAM_EXTERNAL_ID values from environment
```

Comment:
Can you update docs explaining what the IAM_EXTERNAL_ID is for, and how this is intended to be used?